### ISSUE TITLE
Output of `between_residue_bond_loss` has a batch dimension

### ISSUE THREAD

**Issue by hkenlay** [aqlaboratory/openfold#315]
I'm trying to use `openfold.utils.loss.violation_loss` and expecting a scalar as an output, but for the input I'm using I get a vector (matching the batch dimension size). The reason I expected this function to give a scalar is that `openfold.utils.loss.AlphaFoldLoss` iterates over other losses and sums them, and these other losses return scalar values. 

Inside the `openfold.utils.loss.violation_loss` function, the following four quantities are summed to compute the loss
```
loss = (
        violations["between_residues"]["bonds_c_n_loss_mean"]
        + violations["between_residues"]["angles_ca_c_n_loss_mean"]
        + violations["between_residues"]["angles_c_n_ca_loss_mean"]
        + l_clash
    )
```

If we take `violations["between_residues"]["bonds_c_n_loss_mean"]` as an example, this comes from 
```
connection_violations = between_residue_bond_loss(
        pred_atom_positions=atom14_pred_positions,
        pred_atom_mask=batch["atom14_atom_exists"],
        residue_index=batch["residue_index"],
        aatype=batch["aatype"],
        tolerance_factor_soft=violation_tolerance_factor,
        tolerance_factor_hard=violation_tolerance_factor,
    )
```
From `find_structural_violations`. In the return a comment suggests `connection_violations["c_n_loss_mean"]` should be a scalar:
```
    return {
        "between_residues": {
            "bonds_c_n_loss_mean": connection_violations["c_n_loss_mean"],  # ()
```

The `between_residue_bond_loss` has the following inputs 
```
def between_residue_bond_loss(
    pred_atom_positions: torch.Tensor,  # (*, N, 37/14, 3)
    pred_atom_mask: torch.Tensor,  # (*, N, 37/14)
    residue_index: torch.Tensor,  # (*, N)
    aatype: torch.Tensor,  # (*, N)
    tolerance_factor_soft=12.0,
    tolerance_factor_hard=12.0,
    eps=1e-6,
) -> Dict[str, torch.Tensor]:
```

Here is some synthetic inputs showing I get a vector in the output, when I expect a scalar
```
from openfold.utils.loss import between_residue_bond_loss
import torch

batch_size = 32
N = 100

pred_atom_positions = torch.rand(batch_size, N, 14, 3)
pred_atom_mask = torch.randint(0, 2, (batch_size, N, 14))
residue_index = torch.arange(N).unsqueeze(0).repeat(batch_size, 1)
aatype = torch.randint(0, 20, (batch_size, N))
print(
    pred_atom_positions.shape,
    pred_atom_mask.shape,
    residue_index.shape,
    aatype.shape
)
connection_violations = between_residue_bond_loss(
    pred_atom_positions,
    pred_atom_mask,
    residue_index,
    aatype
)
print(connection_violations['c_n_loss_mean'].shape)
```
Output:
```
torch.Size([32, 100, 14, 3]) torch.Size([32, 100, 14]) torch.Size([32, 100]) torch.Size([32, 100])
torch.Size([32])
```

Can anyone spot my misunderstanding? Thanks.
